src/game/draw.c

```diff
@@ -12,10 +12,20 @@
 #include "game/viewport.h"
 #include "global/const.h"
 #include "global/vars.h"
+#include "log.h"
 #include "specific/s_misc.h"
 
 static int16_t m_InterpolatedBounds[6] = { 0 };
 static bool m_CameraUnderwater = false;
+static int32_t m_RoomNumStack[MAX_MATRICES + 1] = { 0 };
```

Comment:
This is of a wrong size, should be `MAX_ROOMS_TO_DRAW`.

Rooms are fine, they don't overflow. What's not fine is when the room drawing pushes matrices infinitely because of bad floor data. We display this stack only to help the builders find out the broken room.